### PR TITLE
Add additional common C++ and Fortran header file extensions

### PR DESCRIPTION
**Pull request by adamjstewart**
See #11588 (@downloadico)

This is a continuation of #10798 (@michaelkuhn)

Can anyone think of any other common header file extensions?

### CODE REVIEW

**Comment by chissg** [comment]
> Can anyone think of any other common header file extensions?

Less common, but still:

C++: `.hxx`

**Comment by downloadico** [comment]
I don't know exactly what version I had checked out!  I did a 'git pull' 
and now things appear to be working!

Thanks,
Jim

On Fri, 31 May 2019, Adam J. Stewart wrote:

> See #11588 (@downloadico)
>
> Can anyone think of any other common header file extensions?
> You can view, comment on, or merge this pull request online at:
>
>  https://github.com/spack/spack/pull/11600
>
> -- Commit Summary --
>
>  * Add additional common C++ and Fortran header file extensions
>
> -- File Changes --
>
>    M lib/spack/llnl/util/filesystem.py (9)
>
> -- Patch Links --
>
> https://github.com/spack/spack/pull/11600.patch
> https://github.com/spack/spack/pull/11600.diff
>
> -- 
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub:
> https://github.com/spack/spack/pull/11600
>

James E. Prewett                    Jim@Prewett.org download@hpc.unm.edu
Systems Team Leader           LoGS: http://www.hpc.unm.edu/~download/LoGS/
Designated Security Officer         OpenPGP key: pub 1024D/31816D93
HPC Systems Engineer III   UNM HPC  505.277.8210

File: lib/spack/llnl/util/filesystem.py [hunk 1]
```diff
@@ -1141,7 +1141,14 @@ def find_headers(headers, root, recursive=False):
         raise TypeError(message)
 
     # Construct the right suffix for the headers
-    suffixes = ['h', 'hpp', 'mod']
+    suffixes = [
+        # C
+        'h',
```
Comment (alalazo) [inline lib/spack/llnl/util/filesystem.py]: OCD: technically `.h` is also common practice in C++ :slightly_smiling_face:

File: lib/spack/llnl/util/filesystem.py [hunk 2]
```diff
@@ -1141,7 +1141,14 @@ def find_headers(headers, root, recursive=False):
...
+        # C++
+        'hpp', 'hxx', 'hh', 'H',
```
Comment (chuckatkins) [inline lib/spack/llnl/util/filesystem.py]: `txx` and `tcc` are common extensions for C++ template implementation headers.

**Comment by chissg** [comment]
Sorry to be nickle-and-diming this one to death, but now it's been mentioned, some communities use `.icc` for template implementations.

**Pull request merged**